src/storage/index/index_manager.cpp

```diff
@@ -0,0 +1,189 @@
+#include "storage/index/index_manager.h"
+#include <memory>
+#include <string>
+#include <vector>
+#include "util/transaction_benchmark_util.h"
+
+namespace terrier::storage::index {
+Index *IndexManager::GetEmptyIndex(transaction::TransactionContext *txn, catalog::db_oid_t db_oid,
+                                   catalog::index_oid_t index_oid, SqlTable *sql_table, bool unique_index,
+                                   const std::vector<std::string> &key_attrs, catalog::Catalog *catalog) {
+  // Setup the oid and constraint type for the index
+  IndexFactory index_factory;
+  ConstraintType constraint = (unique_index) ? ConstraintType::UNIQUE : ConstraintType::DEFAULT;
+  index_factory.SetOid(index_oid);
+  index_factory.SetConstraintType(constraint);
+
+  // Setup the key schema for the index
+  IndexKeySchema key_schema;
+  for (const auto &key_name : key_attrs) {
+    // Get the corresponding column
+    auto col = sql_table->GetSchema().GetColumn(key_name);
```

Comment:
I totally agree with Yuze that someone should catch the out-of-range exception thrown by `GetColumn` because a wrong column name specified by a user should not kill the entire server.

Reply:
Fixed! Thank you.